ratis-server/src/main/java/org/apache/ratis/server/impl/SnapshotInstallationHandler.java

```diff
@@ -68,6 +68,7 @@ class SnapshotInstallationHandler {
     new AtomicReference<>(INVALID_TERM_INDEX);
   private final AtomicBoolean isSnapshotNull = new AtomicBoolean();
   private final AtomicLong installedIndex = new AtomicLong(INVALID_LOG_INDEX);
+  private final AtomicBoolean isSnapshotInstallSuccess = new AtomicBoolean(true);
```

Comment:
Let's use the chunk request index instead of a boolean.

Reply:
Great idea! Great idea! Thank you for your suggestions